File: api-certvalidation/csharp/src/match/crl/CRLMatcher.cs
using tr.gov.tubitak.uekae.esya.api.asn.x509;

namespace tr.gov.tubitak.uekae.esya.api.certificate.validation.match.crl
{
    /**
     * Base class for crl matchers.
     *
     * @author IH
     */
    public abstract class CRLMatcher: Matcher
    {
        protected abstract bool _matchCRL(ECertificate aCertificate, ECRL aCRL);

        /**
         * Sertifika ile SİL'i eşleştirir
         */
        public bool matchCRL(ECertificate aCertificate, ECRL aCRL)
        {
            return (_matchCRL(aCertificate, aCRL));
        }
    }
}

File: api-certvalidation/src/tr/gov/tubitak/uekae/esya/api/certificate/validation/match/crl/CRLMatcher.java
package tr.gov.tubitak.uekae.esya.api.certificate.validation.match.crl;

import tr.gov.tubitak.uekae.esya.api.certificate.validation.match.Matcher;
import tr.gov.tubitak.uekae.esya.api.asn.x509.ECRL;
import tr.gov.tubitak.uekae.esya.api.asn.x509.ECertificate;

/**
 * Base class for crl matchers.
 *
 * @author IH
 */
public abstract class CRLMatcher extends Matcher {

    protected abstract boolean _matchCRL(ECertificate aCertificate, ECRL aCRL);

    /**
     * Sertifika ile SİL'i eşleştirir
     */
    public boolean matchCRL(ECertificate aCertificate, ECRL aCRL) {
        return (_matchCRL(aCertificate, aCRL));
    }

}
